### ISSUE TITLE
az image builder create ignores `vmProfile`

### ISSUE THREAD

**Issue by feiyushi** [Azure/azure-cli#14066]
**Describe the bug**

Bug reported by the author of issue #13972, and verified by me.
When used with `--image-template`, the image template created doesn't contain `vmProfile` node.
Noted that when used with `--debug`, the **json payload of PUT request** doesn't contain the `vmProfile` (though the service will reject the request due to a required missing field that cli has not yet supported).

**To Reproduce**

1. prepare a local image template json payload with `vmProfile` fields
2. az image builder create -g rg -n rn --image-template /path/to/payload --debug
3. the PUT request payload doesn't contain `vmProfile`

**Expected behavior**

Expect that `vmProfile` is included in the image template

**Environment summary**

2.7.0

**Additional context**

**Comment by yungezz** [comment]
hi @qwordy could you pls have a look? thanks

**Comment by qwordy** [comment]
Hi Feiyu, I test it in Azure CLI 2.8.0. `vmProfile` appears in request body. Could you test it again? And could paste the template here?
Log:
```
msrest.http_logger : Request URL: 'https://management.azure.com/subscriptions/1c638cf4-608f-4ee6-b680-c329e824c3a8/resourceGroups/fy/providers/Microsoft.VirtualMachineImages/imageTemplates/template1?api-version=2020-02-14'
msrest.http_logger : Request method: 'PUT'
msrest.http_logger : Request headers:
msrest.http_logger :     'Accept': 'application/json'
msrest.http_logger :     'Content-Type': 'application/json; charset=utf-8'
msrest.http_logger :     'accept-language': 'en-US'
msrest.http_logger :     'Content-Length': '1757'
msrest.http_logger :     'User-Agent': 'python/3.7.4 (Windows-10-10.0.18362-SP0) msrest/0.6.9 msrest_azure/0.6.3 azure-mgmt-imagebuilder/0.4.0 Azure-SDK-For-Python AZURECLI/2.8.0 (PIP)'
msrest.http_logger : Request body:
msrest.http_logger : {"location": "westus", "tags": {"imagebuilderTemplate": "ubuntu1804"}, "properties": {"source": {"type": "PlatformImage", "publisher": "Canonical", "offer": "UbuntuServer", "sku": "18.04-LTS", "version": "18.04.201903060"}, "cu
stomize": [{"name": "RunScriptFromSource", "type": "Shell", "scriptUri": "https://raw.githubusercontent.com/danielsollondon/azvmimagebuilder/master/quickquickstarts/customizeScript.sh"}, {"name": "CheckSumCompareShellScript", "type": "Shell", "scri
ptUri": "https://raw.githubusercontent.com/danielsollondon/azvmimagebuilder/master/quickquickstarts/customizeScript2.sh", "sha256Checksum": "ade4c5214c3c675e92c66e2d067a870c5b81b9844b3de3cc72c49ff36425fc93"}, {"name": "downloadBuildArtifacts", "typ
e": "File", "sourceUri": "https://raw.githubusercontent.com/danielsollondon/azvmimagebuilder/master/quickquickstarts/exampleArtifacts/buildArtifacts/index.html", "destination": "/tmp/index.html"}, {"name": "setupBuildPath", "type": "Shell", "inline
": ["sudo mkdir /buildArtifacts", "sudo cp /tmp/index.html /buildArtifacts/index.html"]}, {"name": "InstallUpgrades", "type": "Shell", "inline": ["sudo apt install unattended-upgrades"]}], "distribute": [{"runOutputName": "runOutputName", "artifact
Tags": {"source": "azVmImageBuilder", "baseosimg": "ubuntu1804"}, "type": "ManagedImage", "imageId": "/subscriptions/1c638cf4-608f-4ee6-b680-c329e824c3a8/resourceGroups/fy/providers/Microsoft.Compute/images/image", "location": "westus"}], "buildTim
eoutInMinutes": 80, "vmProfile": {"vmSize": "Standard_D1_v2", "osDiskSizeGB": 30}}, "identity": {"type": "UserAssigned", "userAssignedIdentities": {"/subscriptions/1c638cf4-608f-4ee6-b680-c329e824c3a8/resourceGroups/fy/providers/Microsoft.ManagedId
entity/userAssignedIdentities/ide1": {}}}}
```
Template:
```
{
    "type": "Microsoft.VirtualMachineImages/imageTemplates",
    "apiVersion": "2020-02-14",
    "location": "westus",
    "dependsOn": [],
    "tags": {
        "imagebuilderTemplate": "ubuntu1804"
    },
    "identity": {
        "type": "UserAssigned",
        "userAssignedIdentities": {
          "/subscriptions/1c638cf4-608f-4ee6-b680-c329e824c3a8/resourceGroups/fy/providers/Microsoft.ManagedIdentity/userAssignedIdentities/ide1": {}
        }
    },
    "properties": {

        "buildTimeoutInMinutes" : 80,
        
        "vmProfile": {
            "vmSize": "Standard_D1_v2",
            "osDiskSizeGB": 30
        },

        "source": {
            "type": "PlatformImage",
                "publisher": "Canonical",
                "offer": "UbuntuServer",
                "sku": "18.04-LTS",
                "version": "18.04.201903060"
        },
        "customize": [
            {
                "type": "Shell",
                "name": "RunScriptFromSource",
                "scriptUri": "https://raw.githubusercontent.com/danielsollondon/azvmimagebuilder/master/quickquickstarts/customizeScript.sh"
            },

            {
                "type": "Shell",
                "name": "CheckSumCompareShellScript",
                "scriptUri": "https://raw.githubusercontent.com/danielsollondon/azvmimagebuilder/master/quickquickstarts/customizeScript2.sh",
                "sha256Checksum": "ade4c5214c3c675e92c66e2d067a870c5b81b9844b3de3cc72c49ff36425fc93"
            },
            
            {
                "type": "File",
                "name": "downloadBuildArtifacts",
                "sourceUri": "https://raw.githubusercontent.com/danielsollondon/azvmimagebuilder/master/quickquickstarts/exampleArtifacts/buildArtifacts/index.html",
                "destination":"/tmp/index.html"
            },

            {
                "type": "Shell",
                "name": "setupBuildPath",
                "inline": [
                    "sudo mkdir /buildArtifacts",
                    "sudo cp /tmp/index.html /buildArtifacts/index.html"
                ]
            },

            {
                "type": "Shell",
                "name": "InstallUpgrades",
                "inline": [
                    "sudo apt install unattended-upgrades"
                ]
            }

        ],
        "distribute": 
            [
                {   "type":"ManagedImage",
                    "imageId": "/subscriptions/1c638cf4-608f-4ee6-b680-c329e824c3a8/resourceGroups/fy/providers/Microsoft.Compute/images/image",
                    "location": "westus",
                    "runOutputName": "runOutputName",
                    "artifactTags": {
                        "source": "azVmImageBuilder",
                        "baseosimg": "ubuntu1804"
                    }
                }
            ]
        }
    }
```

**Comment by feiyushi** [comment]
Hi Feiyue, looks like it's fixed/good in the `2.8.0`. Thanks for looking. Closing the issue.
